src/plugins/merkleMemberships/server/index.ts

```diff
@@ -32,15 +32,116 @@ import { NonEmptyArray } from 'fp-ts/NonEmptyArray';
 import * as NE from 'fp-ts/NonEmptyArray';
 import * as S from 'fp-ts/Semigroup';
 import { FpInterfaceType } from '@lib/plugin/fp/interfaceKind';
+import * as E from 'fp-ts/Either';
+import * as O from 'fp-ts/Option';
+import * as A from 'fp-ts/Array';
+
+export type PublicInputArgs = NonEmptyArray<Field>;
+
+const fieldEncDec: EncodeDecoder<FpInterfaceType, Field> = {
```

Comment:
Decoders/encoders for general use types such as Fields could be moved to a dedicated module

Reply:
I'm not against this but I'd prefer doing it in another pr, since they're only used by the membership plugin at the moment.